src/main/java/codeu/controller/ConversationServlet.java

```diff
@@ -70,7 +71,15 @@ void setConversationStore(ConversationStore conversationStore) {
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response)
       throws IOException, ServletException {
-    List<Conversation> conversations = conversationStore.getAllConversations();
+    String username = (String) request.getSession().getAttribute("user");
+    boolean isRegistered = (username != null);
+    HashSet<Conversation> conversations = null;
+    
+    if (isRegistered) {
```

Comment:
I added some comments about "is_registered" below, but could we just use "username != null" for the check and not have that attribute?